src/univariate/continuous/trapezoidal.jl

```diff
@@ -0,0 +1,115 @@
+"""
+    TrapezoidalDist(a,b,c,d)
+
+The *trapezoidal distribution* with lower limit `a`, upper limit `d`, and a plateau between `b` and `c`. The probability density function is
+
+```math
+f(x; a, b, c, d) = \\begin{cases}
+                     \\frac{2}{d+c-a-b}\\frac{x-a}{b-a}  & \\text{for } a\\le x < b \\\\
+                     \\frac{2}{d+c-a-b}  & \\text{for } b\\le x < c \\\\
+                     \\frac{2}{d+c-a-b}\\frac{d-x}{d-c}  & \\text{for } c\\le x \\le d
+                   \\end{cases}
+```
+
+```julia
+TrapezoidalDist(a, b, c, d)
+
+params(d)       # Get the parameters, i.e. (a, b, c, d)
+minimum(d)      # Get the lower bound, i.e. a
+maximum(d)      # Get the upper bound, i.e. d
+```
+
+External links
+
+* [Trapezoidal distribution on Wikipedia](https://en.wikipedia.org/wiki/Trapezoidal_distribution)
+
+"""
+struct TrapezoidalDist{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+    c::T
+    d::T
+    TrapezoidalDist{T}(a::T, b::T, c::T, d::T) where {T <: Real} = new{T}(a, b, c, d)
+end
+
+
+function TrapezoidalDist(a::T, b::T, c::T, d::T; check_args=true) where {T <: Real}
+    check_args && @check_args(TrapezoidalDist, a <= b <= c <= d)
+    return TrapezoidalDist{T}(a, b, c, d)
+end
+
+
+TrapezoidalDist(a::Real, b::Real, c::Real, d::Real) = TrapezoidalDist(promote(a, b, c, d)...)
+TrapezoidalDist(a::Integer, b::Integer, c::Integer, d::Integer) = TrapezoidalDist(float(a), float(b), float(c), float(d))
+
+@distr_support TrapezoidalDist d.a d.d
+
+#### Conversions
+
+convert(::Type{TrapezoidalDist{T}}, a::Real, b::Real, c::Real, d::Real) where {T<:Real} = TrapezoidalDist(T(a), T(b), T(c), T(d))
+convert(::Type{TrapezoidalDist{T}}, d::TrapezoidalDist{S}) where {T<:Real, S<:Real} = TrapezoidalDist(T(d.a), T(d.b), T(d.c), T(d.d), check_args=false)
+
+#### Parameters
+
+params(d::TrapezoidalDist) = (d.a, d.b, d.c, d.d)
+partype(::TrapezoidalDist{T}) where {T<:Real} = T
+
+
+#### Statistics
+
+function mean(d::TrapezoidalDist)
+    ((d.d^3 - d.c^3)/(d.d-d.c) - (d.b^3 - d.a^3)/(d.b-d.a)) / (3*(d.d+d.c-d.b-d.a))
+end
+
+function var(d::TrapezoidalDist)
+    ((d.d^4 - d.c^4)/(d.d-d.c) - (d.b^4 - d.a^4)/(d.b-d.a)) / (6*(d.d+d.c-d.b-d.a)) - mean(d)^2
+end
+
+function entropy(d::TrapezoidalDist{T}) where {T<:Real}
+    (a, b, c, d) = params(d)
+    (d-c+b-a)/(2*(d+c-b-a)) + log(d+c-b-a) - log(2)
+end
+
+#### Evaluation
+
+function pdf(d::TrapezoidalDist{T}, x::Real) where T<:Real
+    (a, b, c, d) = params(d)
+    x <= a ? zero(T) :
+        a <= x <  b ? 2 / (d+c-a-b) * (x-a)/(b-a) :
+        b <= x <  c ? 2 / (d+c-a-b) :
+        c <= x <= d ? 2 / (d+c-a-b) * (d-x)/(d-c) : zero(T)
+end
+
```

Comment:
Probably the only thing missing is:
```suggestion
logpdf(d::TrapezoidalDist, x::Real) = log(pdf(d, x))
```

BTW `pdf` (and `cdf` etc.) are not type stable for, e.g., `T = Rational{Int}` or `T = Float32` and `typeof(x) = Float64`.

Reply:
Perhaps we can do `clamp(min((x-a)/(b-a), (d-x)/(d-c), 0, 1)`

Reply:
Yes, it seems not to make sense that the return type of `pdf` depend on the type of the distribution parameters `T`. It should rather depend on the type of `x`. I think `clamp` will do the trick, I will have a look.

 PS: the current implementation follows closely the triangle distribution, so we may want to change this too.